contracts/Chain.sol

```diff
@@ -49,7 +50,10 @@ contract Chain is ChainConfig, ReentrancyGuard {
     // unfortunately we can't be able to delete this data to release gas, why?
     // because to do this, we need to save all the keys and then run loop for all keys... that may cause OOG
     // also storing keys is more gas consuming so... I made decision to stay with mapping and never delete history
-    mapping(uint256 => mapping(bytes32 => uint256)) counts;
+    mapping (uint256 => mapping(bytes32 => uint256)) counts;
+
+    /// @dev shard => total amount of tokens
+    mapping (uint256 => uint256) tokens;
```

Comment:
what about shards instead of tokens....

Reply:
or balancesPerShard kinda what you called it below

Reply:
hmm why a struct? do we need more data than balance? please explain.

Reply:
your right struct might be an overkill. I think having a struct makes it easier to read and removes the need for comment.